xarray/tests/test_dataset.py

```diff
@@ -6694,6 +6694,16 @@ def test_polyfit_weighted(self) -> None:
         ds.polyfit("dim2", 2, w=np.arange(ds.sizes["dim2"]))
         xr.testing.assert_identical(ds, ds_copy)
 
+    def test_polyfit_coord(self) -> None:
```

Comment:
Can you add an explicit test for expected output please? e.g. the example from #4375

Reply:
Also done!